extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisReceiver.scala

```diff
@@ -120,16 +105,27 @@ private[kinesis] class KinesisReceiver(
    */
   override def onStart() {
     workerId = Utils.localHostName() + ":" + UUID.randomUUID()
-    credentialsProvider = new DefaultAWSCredentialsProviderChain()
-    kinesisClientLibConfiguration = new KinesisClientLibConfiguration(appName, streamName,
-      credentialsProvider, workerId).withKinesisEndpoint(endpointUrl)
+
+    // KCL config instance
+    val kinesisClientLibConfiguration = new KinesisClientLibConfiguration(appName, streamName,
+       resolveAWSCredentialsProvider(), workerId).withKinesisEndpoint(endpointUrl)
```

Comment:
All this formatting seems messed up. This PR requires a huge formatting overhaul.